ang/civicase/ActivitiesCalendar.js

```diff
@@ -407,15 +463,27 @@
     }
 
     /**
-     * Loads the dates with at least an activity with the given status(es)
+     * Loads the days within the month of the given date
+     * with at least an activity with the given status(es)
+     *
+     * The days are returned in an object containing also the year+month they
+     * belong to, so that they can be properly grouped in the internal list of days
      *
-     * @param {*} statusParam
+     * @param {*} status
+     * @param {Date} date
      * @return {Promise}
      */
-    function loadDaysWithActivities (statusParam) {
+    function loadDaysWithActivities (status, date) {
       var params = {};
+      var dateMoment = moment(date);
 
-      params.status_id = statusParam;
+      params.status_id = status;
+      params.activity_date_time = {
+        BETWEEN: [
+          dateMoment.startOf('month').format('YYYY-MM-DD') + ' 00:00:00',
+          dateMoment.endOf('month').format('YYYY-MM-DD') + ' 23:59:59'
```

Comment:
when using `startOf` and `endOf` there's no need to specify the end date and time. this can be:

```js
BETWEEN: [
  dateMoment.startOf('month').format('YYYY-MM-DD HH:mm:ss'),
  dateMoment.endOf('month').format('YYYY-MM-DD HH:mm:ss')
]
```

[More info here](https://momentjs.com/docs/#/manipulating/start-of/)

Reply:
cool, didn't know it, thanks